be/src/io/cache/block_file_cache.h

```diff
@@ -446,6 +489,9 @@ class BlockFileCache {
     LRUQueue _disposable_queue;
     LRUQueue _ttl_queue;
 
+    // keys for async remove
+    std::shared_ptr<boost::lockfree::spsc_queue<FileCacheKey>> _recycle_keys;
```

Comment:
it seems there is no contention on _recycle_keys.
it is better to  use regular data structure instead for better understanding and maintainance